cmd/traffic/cmd/manager/mutator/agent_injector.go

```diff
@@ -204,7 +204,11 @@ func (a *agentInjector) Uninstall(ctx context.Context) {
 	a.agentConfigs.DeleteMapsAndRolloutAll(ctx)
 }
 
-func needInitContainer(config *agentconfig.Sidecar) bool {
+func needInitContainer(ctx context.Context, config *agentconfig.Sidecar) bool {
+	if !config.InitContainerEnabled {
```

Comment:
```suggestion
	if !managerutil.GetEnv(ctx).AgentInitContainerEnabled {
```
That way, the attribute can be remove from the config altogether.